priv/static/phoenix_live_reload.js

```diff
@@ -6,96 +6,203 @@
+  }
 
-  return newLink;
-};
+  newLink.onerror = onComplete
+  newLink.onload  = onComplete
+  link.setAttribute("data-pending-removal", "")
+  newLink.setAttribute("rel", "stylesheet");
+  newLink.setAttribute("type", "text/css");
+  newLink.setAttribute("href", url + (url.indexOf("?") >= 0 ? "&" : "?") + "vsn=" + date)
+  link.parentNode.insertBefore(newLink, link.nextSibling)
+  return newLink
+}
 
-var repaint = function(){
-  var browser = navigator.userAgent.toLowerCase();
-  if(browser.indexOf('chrome') > -1){
-    setTimeout(function(){ document.body.offsetHeight; }, 25);
+let repaint = () => {
+  let browser = navigator.userAgent.toLowerCase()
+  if(browser.indexOf("chrome") > -1){
+    setTimeout(() => document.body.offsetHeight, 25)
   }
-};
+}
 
-var cssStrategy = function(){
-  var reloadableLinkElements = window.parent.document.querySelectorAll(
-    'link[rel=stylesheet]:not([data-no-reload]):not([data-pending-removal])'
-  );
+let cssStrategy = () => {
+  let reloadableLinkElements = window.parent.document.querySelectorAll(
+    "link[rel=stylesheet]:not([data-no-reload]):not([data-pending-removal])"
+  )
 
-  [].slice
-    .call(reloadableLinkElements)
-    .filter(function(link) { return link.href })
-    .forEach(function(link) { buildFreshUrl(link) });
+  ([].slice).call(reloadableLinkElements)
+    .filter(link => link.href)
+    .forEach(link => buildFreshUrl(link))
 
-  repaint();
-};
+  repaint()
+}
 
-var pageStrategy = function(chan){
-  chan.off('assets_change');
-  window[targetWindow].location.reload();
-};
+let pageStrategy = channel => {
+  channel.off("assets_change")
+  window[targetWindow].location.reload()
+}
 
-var reloadStrategies = {
+let reloadStrategies = {
   js: jsStrategy,
   css: reloadPageOnCssChanges ? pageStrategy : cssStrategy,
   page: pageStrategy
 };
 
-socket.connect();
-var chan = socket.channel('phoenix:live_reload', {})
-chan.on('assets_change', function(msg) {
-  var reloadStrategy = reloadStrategies[msg.asset_type] || reloadStrategies.page;
-  setTimeout(function(){ reloadStrategy(chan, msg); }, interval);
-});
-chan.join().receive("ok", () => console.log("[PhoenixLiveReload] Connected!"));
+class LiveReloader {
+  constructor(socket){
+    this.socket = socket
+    this.logsEnabled = false
+    this.enabledOnce = false
+    this.editorURL = null
+  }
+  enable(){
+    this.socket.onOpen(() => {
+      if(this.enabledOnce){ return }
+      this.enabledOnce = true
+      if(["complete", "loaded", "interactive"].indexOf(parent.document.readyState) >= 0){
+        this.dispatchConnected()
+      } else {
+        parent.addEventListener("load", () => this.dispatchConnected())
+      }
+    })
+
+    this.channel = socket.channel("phoenix:live_reload", {})
+    this.channel.on("assets_change", msg => {
+      let reloadStrategy = reloadStrategies[msg.asset_type] || reloadStrategies.page
+      setTimeout(() => reloadStrategy(this.channel), interval)
```

Comment:
多分こうですかね？ここまでmsgは正しく渡ってますがここでjsStrategyに渡されずにundefinedになってるかもです

```suggestion
      setTimeout(() => reloadStrategy(this.channel, msg), interval)
```

Reply:
私もそこだと思い、確認中です…！ありがとうございます！

Reply:
対応しました！live reloadも確認しました

c8e55cb8d93de8900e9dbc546be7da120765bd8e